script/tool/lib/src/common/repository_package.dart

```diff
@@ -142,6 +142,23 @@ class RepositoryPackage {
       !isPlatformInterface &&
       directory.basename != directory.parent.basename;
 
+  /// True if this appears to be an example package, according to package
+  /// conventions.
+  bool get isExample {
+    final RepositoryPackage? enclosingPackage = getEnclosingPackage();
+    if (enclosingPackage == null) {
+      // An example package is enclosed in another package.
+      return false;
+    }
+    for (final RepositoryPackage example in enclosingPackage.getExamples()) {
+      if (example.path == path) {
+        // This package is an example of its enclosing package.
+        return true;
+      }
+    }
+    return false;
```

Comment:
Optional: Everything from 153 to here could collapse to:
```dart
// Check whether this is one of the enclosing package's examples.
return enclosingPackage.getExamples().any((RepositoryPackage p) => p.path == path);
```

Reply:
Oooo this is neat!